### PR TITLE
Various improvements

### PR DESCRIPTION
**Pull request by bmigirl**
- improve radio boxes styling (add click on a whole radio box, add styling accordingly)
- add auto country selection in billing address based on shipping address

### CODE REVIEW

**Review by orzechdev — APPROVED** [review]

File: apps/checkout/src/providers/CountrySelectProvider.tsx [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  setCountryCodeFromAddress: (address?: AddressFragment | null) => void;
+}
+
+export const [useCountrySelect, Provider] =
+  createSafeContext<CountrySelectContextConsumerProps>();
+
+export const CountrySelectProvider: React.FC<
+  PropsWithChildren<CountrySelectProviderProps>
+> = ({ children, selectedCountryCode }) => {
+  const [countryCode, handleSetCountryCode] = useState<CountryCode>(
```
Comment (Cloud11PL) [inline apps/checkout/src/providers/CountrySelectProvider.tsx:23]: State `countryCode` and prop `setCountryCode` names are confusing. Maybe the prop should be named with prefix `handler`?

**Review by Cloud11PL — CHANGES_REQUESTED** [review]

File: apps/checkout/src/providers/CountrySelectProvider.tsx (hunk 1 above)
Comment (bmigirl) [reply apps/checkout/src/providers/CountrySelectProvider.tsx:23]: I don't think I understand, what's confusing here?

**Review by Cloud11PL — APPROVED** [review]

File: apps/checkout/src/sections/Addresses/Addresses.tsx [hunk 2]
```diff
@@ -22,18 +23,30 @@ export const Addresses: React.FC = () => {
...
-        <ShippingAddressSection
+        <CountrySelectProvider
+          selectedCountryCode={
+            checkout?.shippingAddress?.country?.code as CountryCode
+          }
+        >
+          <ShippingAddressSection
+            addresses={userAddresses}
+            defaultShippingAddress={user?.defaultShippingAddress}
+            useShippingAsBillingAddress={useShippingAsBillingAddress}
```
Comment (jwm0) [inline apps/checkout/src/sections/Addresses/Addresses.tsx:34]: Might be worth renaming it so it doesn't sound like a hook. 'isBillingSameAsShippingAddress'

**Review by jwm0 — APPROVED** [review]

File: apps/checkout/src/sections/Addresses/Addresses.tsx (hunk 2 above)
Comment (bmigirl) [reply apps/checkout/src/sections/Addresses/Addresses.tsx]: Done

**Pull request merged**